lib/core/src/client/preview/start.tsx

```diff
@@ -0,0 +1,95 @@
+import { navigator, window } from 'global';
+
+import addons, { DecorateStoryFunction, Channel } from '@storybook/addons';
+import createChannel from '@storybook/channel-postmessage';
+import { ClientApi, ConfigApi, StoryStore } from '@storybook/client-api';
+import Events from '@storybook/core-events';
+
+import { initializePath, setPath } from './url';
+import { RenderStoryFunction } from './types';
+import { makeConfigure } from './makeConfigure';
+import { StoryRenderer } from './StoryRenderer';
+
+const isBrowser =
+  navigator &&
+  navigator.userAgent &&
+  navigator.userAgent !== 'storyshots' &&
+  !(navigator.userAgent.indexOf('Node.js') > -1) &&
+  !(navigator.userAgent.indexOf('jsdom') > -1);
+
+function getOrCreateChannel() {
+  let channel = null;
+  if (isBrowser) {
+    try {
+      channel = addons.getChannel();
+    } catch (e) {
+      channel = createChannel({ page: 'preview' });
+      addons.setChannel(channel);
+    }
+  }
+
+  return channel;
+}
+
+function getClientApi(channel: Channel, decorateStory: DecorateStoryFunction) {
```

Comment:
`channel` can be `null` here.

PS Can you clarify, why channel created only in browser?

Reply:
I'm not 100% sure, but "not browser" in this file means storyshots.

(This file is not new, it just got moved from a js file, btw).

I don't really know why storyshots doesn't create a channel. I suppose it is technically not possible to open a postmessage channel in jest/node? In any case that is the state of the world as things stand. One refactor I think someone should do is to make SS create a mock channel so we don't need to do these checks.

Reply:
Oh I see you were saying I should make it nullable; good call!

Reply:
https://github.com/storybookjs/storybook/pull/9914/commits/4c52896bdeb0600324b12939def0106d280c33ba

Reply:
Hm... I init storybook inside nodejs in my project https://github.com/wKich/creevey and channel works fine, maybe because I initiate jsdom as well. I need `SET_STORIES` event to get stories meta, but with that check no other way to pass channel inside `StoryStore` and `ClientApi` even `addons.getChannel();` return channel without exception.